lib/truffle/truffle/cext.rb

```diff
@@ -1173,8 +1173,8 @@ def rb_define_class_under(mod, name, superclass)
       raise ArgumentError, "no super class for `#{name}'"
     end
 
-    if mod.const_defined?(name, false)
-      current_class = mod.const_get(name, false)
+    if Primitive.module_const_defined?(mod, name, false, false)
+      current_class = Primitive.module_const_get(mod, name, false, true, false)
```

Comment:
I don't follow. The `true` value for the `look_in_object` parameter is hard-coded in https://github.com/oracle/truffleruby/blob/12f0d6984ee4f0119b55755c479d31f2ec809556/src/main/ruby/truffleruby/core/module.rb#L108 and that's the call I was replacing.

Reply:
`look_in_object` has no meaning with inherit=false, so ultimately it doesn't matter, and we get into `getConstantNoInherit` with a Symbol.
In `Module#const_get` it's giving `look_in_object=true` for the case `inherit` is true, and if it's not true just relies on it being ignored.

So here semantically because we know inherit=false, it makes more sense to use look_in_object=false since we won't look in Object.

Reply:
FWIW I was looking at https://github.com/oracle/truffleruby/blob/12f0d6984ee4f0119b55755c479d31f2ec809556/src/main/ruby/truffleruby/core/module.rb#L130-L131 and so thought it was obvious but indeed it's not obvious due to the module_const_get before.